tests/integration/conftest.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+import boto3
+import boto3.session
+import pytest
+import yaml
+from botocore.client import Config
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+TEST_BUCKET_NAME = "kyuubi-test"
+TEST_NAMESPACE = "kyuubi-test"
+TEST_SERVICE_ACCOUNT = "kyuubi-test"
+TEST_POD_SPEC_FILE = "./tests/integration/setup/testpod_spec.yaml"
+
+
+class MockCharm(BaseModel):
```

Comment:
**question** Why is this called `MockCharm`? This does not seem mock (like a test charm) but rather some charm metadata for the version. I checked on Spark History Server, and indeed, we call this similar class `CharmVersion` (see [here](https://github.com/canonical/spark-history-server-k8s-operator/blob/main/tests/integration/conftest.py#L10)).

Reply:
`CharmVersion` name seemed a bit off to me because the first time I read the name I mistook the class is about a version. Maybe `MockCharm` could be replaced with `TestCharm`, just so that we refrain from using the word "mock" (which in this case it is not).

Reply:
yes, naming is a mess. Let's discuss this later to find a suitable name we all like :D

Reply:
This has been updated to `TestCharm`.